Sprint 15 (home work) Gr26

amanzhola opened the pull request:
👉 (📱) ➡️ 🧠 🔄 ➡️ 💾

amanzhola committed:
Sprint 15 (home work) Gr26
👉 (📱) ➡️ 🧠 🔄 ➡️ 💾

amanzhola committed:
Sprint 15 (home work) Gr26
🎵🧼🏗️✅
🎬🧼🏗️✅
🌤️🧼🏗️✅
+ 👉 ☀️ 🔁 🌙 (🧼🏗️✅)

sukhoikms27 commented on app/src/main/java/com/example/playlistmaker/creator/Creator.kt:
```diff
@@ -0,0 +1,109 @@
+    }
+
+    // --- Метод для предоставления AudioInteraction с учетом SearchHistoryRepository ---
+    fun provideAudioInteraction(context: Context): AudioInteraction {
```
context можно передать в `Creator` на старте приложения, тогда не потребуется передавать его во входные параметры, а напрямую использовать для создания объектов здесь

sukhoikms27 commented on app/src/main/java/com/example/playlistmaker/ui/audio/SearchActivity.kt:
```diff
@@ -102,30 +97,31 @@ class SearchActivity : BaseActivity(), OnTrackClickListener {
-                        if (isInHistory) searchHistory.showHistory()
-                        else searchHistory.hideHistory()
+                        if (isInHistory) {
+                            history.visibility = VISIBLE
```
У `view` есть удобный экстеншен: `View.isVisible = true/false`, которая заменяет привычное нам `View.visibility = View.VISIBLE/View.GONE`

sukhoikms27 commented:
Экраны в приложении:
<img width="311" alt="image" src="https://github.com/user-attachments/assets/da92f80b-f90c-4733-8dbf-a3ee2f4c2bc3" />
<img width="309" alt="image" src="https://github.com/user-attachments/assets/00bcb82d-0a14-4510-9247-7225a3b62b35" />
<img width="313" alt="image" src="https://github.com/user-attachments/assets/bf04dd0c-b8bb-44bc-a388-908565c7ac44" />

Экраны по макетам:
<img width="244" alt="image" src="https://github.com/user-attachments/assets/cd99327d-fc84-4454-9d38-670781cadb01" />
<img width="269" alt="image" src="https://github.com/user-attachments/assets/71932ec5-a5e1-4015-adce-da72e155e80f" />
<img width="213" alt="image" src="https://github.com/user-attachments/assets/7a31d24a-9ee6-415b-bc75-1d1136286ed3" />

amanzhola committed:
Sprint 15 (home work) Gr26
+ 👉 📻(🎶)🧼🏗️✅

amanzhola committed:
Sprint 15 (home work) Gr26
1) Внешний вид - Не понятно, если чел то увидит(не глазами) она перелючается. Если ИИ то не поймет (тогда скрыл, для чела открыто)
2) сделано
3) сделано
4) Замечение написано туманно, в скобках было такое (работа с темой) -> возможно это в SettingsActivity тогда сделано

Доп без обяснении:
❌ Приложение работает корректно: возможности, предоставляемые приложением до рефакторинга, предоставляются в том же объёме и качестве.

Здесь аут: Приложение как работала до так и работает после рефакторинга. Нет каких либо обьяснении или замечанию или хоть что-то мин показывающее, что приложение стала работать лучше или хуже чем до рефакторинга.

amanzhola committed:
Sprint 15 (home work) Gr26

Доп без обяснении:
❌ Приложение работает корректно: возможности, предоставляемые приложением до рефакторинга, предоставляются в том же объёме и качестве.

Дошло(наконец-то секрет вашей тайнственности в молчани!!!): извините, реально ни разу не видел, только увидел красное сердечко для игры. Ну это 5 или ну предел 15 минут(сделано)

amanzhola committed:
Sprint 15 (home work) Gr26

Доп без обяснении:
❌ Приложение работает корректно: возможности, предоставляемые приложением до рефакторинга, предоставляются в том же объёме и качестве.

+ И ты снова прав (оказалось ухудшилась, я думал улучшилась), пока смотрел сердце чтобы было во время песни обнаружил, авто-выключение оказалось  потерялось при переходе. Прелесть чистой архитекрутры, как раз можно тестить не переделывая весь код и найти что пропустил при переходе, тем более в 14 все работал 100% (вернул обратно-сделано)
Это заняло 5 часов, у меня она работала 13 и 14 поэтому думал легко и перенесу быстро, из-за этого и заняло столько времени. Пришлось каждую строку переносить по буквам

По 2м вашим пунктам, что по ходу добавили сверху допом:

1 - кнопка изи потому, что SearchHistoryInteraction как раз в вьюмодели, так что это минут 10
2 - убрать шаре тоже 10 минут. Я думал что он же как String, Int, прочие из-за названия SharedPreference и значит у него преференция. Ок тоже легко ведь опять же SearchHistoryInteraction как раз в вью модели. Я на шаре напал потому что легко когда любые измения в памяти и ты сразу в курсе - очень удобная вещь.
Ну добавочный Ваш про 3 раза: сделал все ваши пункты + добавочно которое нашел по ходу, что заняло больше всего времени. Все ваши комменты легко решаются, главное указать как вы сделали на этот раз. Узнал, что Шаре нет преференции, потом узнал что делаю правильно ведь наугад же (все впервые в жизни) только благодаря Практикуму и делал, он реально красавчик, вот бы всех убрать кроме него, но говорить он и не умеет и на мелкие как типа кнопки когда у тебя уже проведен SearchHistoryInteraction и прочее это называется опыт и его никакими книгами не получить к сожалению и вот так она приходит как сейчас.

amanzhola committed:
Sprint 15 (home work) Gr26

ой пока по каждой букве переносил с 13 и 14 тамже пришлось прямо по букве переносить, там у меня сложная логика. И забыл 3 часика, убрал сейчас. Жалко если воспользуетесь моей оплошностью.

Сейчас проверял, потому что sharepref классная штука, но опасная, она же реагирует на изменения и как раз может очистить память после уже так как выйти из апа, она сидить же внутри вьюмодели и жизнь у нее макс. И увидел эти часики блин.

amanzhola committed:
Sprint 15 (home work) Gr26

✅  улучшен аудио.
✂️ Убран AudioPlayerInteractionImpl из активити которые оказались минуя Creator

amanzhola committed:
Sprint 15 (home work) Gr26

Наконец-то 100% использован Creator для SearchActivity and ExtraOption(Final)

amanzhola committed:
Sprint 15 (home work) Gr26
RU (по умолчанию):
a) 🧢🔒 — тулбар (но при нажатии: 🔴🎯)
b) 🧩✅ — main
c) 📝🚫😊 — строки

EN:
a) 🧢🔓🔴 — тулбар, (dropdown, filter)
b) 🧩❌➡️😄 — в main
c) 📝🔁❤️😄 — в строках

amanzhola committed:
Sprint 15 (home work) Gr26
Благодаря "хлебному месту" Леонида Сухинина в группе Alexander Hamilton которая оказалось похлеще Красного Хлеба так что чуть пришлось подредактировать не только sw320dp и sw360dp, а даже sw393dp

amanzhola committed:
Sprint 15 (home work) Gr26
🧩 BaseActivity 💥 ➡️ (👨‍🔧 ➕ 🧰)

amanzhola committed:
Sprint 15 (home work) Gr26
🧩 BaseActivity 🔚 🤔 🧼 🧹 🏗
🧩 SettingsActivity 🛠 ➡️ ♻️ 🧼 🏗
🧩 MainActivity✳️ 🎯 ➡️ ♻️ 🧼 🏗

amanzhola committed:
Sprint 15 (home work) Gr26
change or add: 65 files
🧩 SearchActivity ♻️
🧩 SearchViewModel ♻️
🧩 SearchMovie ♻️
🧩 ExtraOption ♻️
🧩 ExtraOptionViewModel ♻️
🧩 LauncherActivity ♻️
🧩 MoviePager ♻️
🧩 MoviePagerList ♻️
🧩 MoviesAdapterList ♻️
🧩 TrackAdapter ♻️
🧩 TrackDetailActivity ♻️
🧩 TrackPreviewActivity ♻️
🧩 TrackPreviewViewModel ♻️

✳️ 🎯 ➡️ ♻️ 🧼 🏗
xml -> activity_movie_pager.xml, activity_search.xml, i_movie1.xml, move_down_24.xml, strings.xml, strings(en).xml,themes.xml

✳️ 🎯 ➡️ ♻️ 🧼 🏗
Agreement.kt, AgreementImpl, App.kt, AppShareHelper.kt, AudioErrorManager.kt, AudioSingleTrackImpl, AudioSingleTrackShare.kt, AudioTracksImpl.kt, AudioTracksShare.kt, BottomNavigationHelper.kt, CheckInternetConnectionUseCase.kt, ColorApplierHelper.kt, Creator.kt, GsonMovieSerializer.kt, GsonTrackSerializer.kt, JsonFileWriter.kt, LegalHelper.kt, MovieSerializer.kt, MovieStorageHelper.kt, NetworkChecker.kt, NetworkRepository.kt, NetworkRepositoryImpl, NetworkStatusChecker.kt, NetworkStatusCheckerImpl, NetworkUtils.kt, ResourceColorProvider.kt, ResourceColorProviderImpl, SearchUiState.kt, SegmentManager.kt, SegmentTextHelper.kt, Share.kt, SharedPrefsMovieStorage.kt, SharedPrefsTrackStorage.kt, ShareImpl.kt, ShareMovie.kt, ShareMovieImpl.kt, Support.kt, SupportHelper.kt, SupportImpl.kt, TrackListInputData.kt, TrackListIntentParser.kt, TrackListIntentParser.kt, TrackListIntentParserImpl.kt, TrackSerializer.kt, TrackStorageHelper.kt

➕ 🎥➡️⬅️🎥 ➡️ (🆕 🎬 Обмен фильмами доступен )
➕ 🌺🔄 🌼🔃 ➡️ (✨ 🌸 Цветок на тулбаре теперь вращается)

🔚 🤔 🧼 🧹 🏗

amanzhola committed:
Sprint 15 (home work) Gr26
🔧✅✏️📴🔻🆙🔘(fixed update btn underline)

amanzhola committed:
Sprint 15 (home work) Gr26
⭐ 📽️ 🍿 🎬 ✨ 💖 🔝 (Movie 💖 done)
+ tx 🚀🔥 Vlamidir(🐞➖ ➡️ 🚫 ➡️ 😎)

amanzhola committed:
Sprint 16 (home work) Gr26
